samcli/lib/list/testable_resources/testable_resources_to_table_mapper.py

```diff
@@ -0,0 +1,67 @@
+"""
+Implementation of the testable resources to table mapper
+"""
+from typing import Dict, Any
+from collections import OrderedDict
+from samcli.lib.list.list_interfaces import Mapper
+
+
+class TestableResourcesToTableMapper(Mapper):
+    """
+    Mapper class for mapping testable-resources data for table output
+    """
+
+    def map(self, data: list) -> Dict[Any, Any]:
+        """
+        Maps data to the format needed for consumption by the table consumer
+
+        Parameters
+        ----------
+        data: list
+            List of dictionaries containing the entries of the testable resources data
+
+        Returns
+        -------
+        table_data: Dict[Any, Any]
+            Dictionary containing the information and data needed for the table consumer
+            to output the data in table format
+        """
+        entry_list = []
+        for testable_resource in data:
+            cloud_endpoint_furl_string = testable_resource.get("CloudEndpointOrFunctionURL", "-")
+            methods_string = "-"
+            cloud_endpoint_furl_multi_list = []
+            if isinstance(testable_resource.get("CloudEndpointOrFunctionURL", "-"), list) and testable_resource.get(
+                "CloudEndpointOrFunctionURL", []
+            ):
+                cloud_endpoint_furl_string = testable_resource.get("CloudEndpointOrFunctionURL", ["-"])[0]
+                if len(testable_resource.get("CloudEndpointOrFunctionURL", [])) > 1:
+                    cloud_endpoint_furl_multi_list = testable_resource.get("CloudEndpointOrFunctionURL", ["", ""])[1:]
+            if isinstance(testable_resource.get("Methods", "-"), list) and testable_resource.get("Methods", []):
+                methods_string = "; ".join(testable_resource.get("Methods", []))
+
+            entry_list.append(
+                [
+                    testable_resource.get("LogicalResourceId", "-"),
+                    testable_resource.get("PhysicalResourceId", "-"),
+                    cloud_endpoint_furl_string,
+                    methods_string,
+                ]
+            )
+            if cloud_endpoint_furl_multi_list:
+                for url in cloud_endpoint_furl_multi_list:
+                    entry_list.append(["", "", url, ""])
+        table_data = {
+            "format_string": "{Resource ID:<{0}} {Physical ID:<{1}} {Cloud Endpoint/FURL:<{2}} {Methods:<{3}}",
+            "format_args": OrderedDict(
+                {
+                    "Resource ID": "Resource ID",
+                    "Physical ID": "Physical ID",
+                    "Cloud Endpoint/FURL": "Cloud Endpoint/Function URL",
+                    "Methods": "Methods",
+                }
+            ),
+            "table_name": "Testable Resources",
```

Comment:
Somewhat tangential, but this concept of "testable resource" isn't intuitive for me, and it's not clear why this should be a separate subcommand versus an option to `sam list`.

Perhaps it can be made more intuitive by thinking what _really_ is the difference with them and all other resources. I would argue that _all_ resources are testable since I can interact with them through HTTP.

Reply:
Agreed. This naming probably isn't the best. We're currently in discussion with UX and Molly to come up with something better, but we're thinking something along the lines of `sam list endpoints`.